core/services/ocr2/plugins/ocr2keeper/evm/registry.go

```diff
@@ -522,28 +522,33 @@ func (r *EvmRegistry) doCheck(ctx context.Context, keys []types.UpkeepKey, chRes
 // TODO (AUTO-2013): Have better error handling to not return nil results in case of partial errors
 func (r *EvmRegistry) checkUpkeeps(ctx context.Context, keys []types.UpkeepKey) ([]types.UpkeepResult, error) {
 	var (
-		checkReqs    = make([]rpc.BatchElem, len(keys))
-		checkResults = make([]*string, len(keys))
+		multiErr     error
+		results      []types.UpkeepResult
+		checkReqs    []rpc.BatchElem
+		checkResults []*string
```

Comment:
For slight performance gains you should set the capacity of the arrays you will be appending to. The previous version set the length equal to the number of keys and the loop set the value at each index. That is the most performant, but when you cannot do that and need to append, you should set the length to 0 and the capacity to the maximum you're likely to use. This will keep the go runtime from trying to guess and resizing the array as it grows.

`checkReqs = make([]rpc.BatchElem, 0, len(keys))`

Reply:
Okay, sounds good, I normally only set capacity on slices for really large datasets since I find the performance difference pretty negligible otherwise, but I'll flip this back to use the capacity and stick with appending 🙌